### DISCUSSION TITLE
how to use readonly to query sql and write data at same time

### DISCUSSION THREAD

**Discussion by xubobbs** [duckdb/duckdb#14341] [Q&A]
hi, 
I use go duckdb to make some work .
in background , I use read data from other database and write it in duckdb .
and at same time, maybe need do some query ,I use READ ONLY MODE to open the same duckdb db file .

so I often find query in read only and write the db at same time , sysetm will often crash because of segment execption .

so What can I do ?

**Comment by xubobbs** [comment]
and crash in 
```
Exception has occurred: fatal error
Error getting throw reason: could not find symbol value for s
Stack:
	24  0x00000000045976eb in C._cgo_65c6d7ab71cf_Cfunc_duckdb_execute_pending
	     at /tmp/go-build/cgo-gcc-prolog:431
	28  0x0000000003fdd6cb in github.com/marcboeker/go-duckdb._Cfunc_duckdb_execute_pending
	     at _cgo_gotypes.go:1107
	29  0x0000000003ff7116 in github.com/marcboeker/go-duckdb.(*stmt).execute.func6
	     at /home/xubo/go/pkg/mod/github.com/marcboeker/go-duckdb@v1.7.0/statement.go:245
	30  0x0000000003ff6c4a in github.com/marcboeker/go-duckdb.(*stmt).execute
	     at /home/xubo/go/pkg/mod/github.com/marcboeker/go-duckdb@v1.7.0/statement.go:245
	31  0x0000000003ff66bd in github.com/marcboeker/go-duckdb.(*stmt).QueryContext
	     at /home/xubo/go/pkg/mod/github.com/marcboeker/go-duckdb@v1.7.0/statement.go:200
	32  0x0000000001adcb3f in database/sql.ctxDriverStmtQuery
```

**Comment by Tishj** [comment]
That doesn't sound like that is supported currently.
<https://duckdb.org/docs/connect/concurrency.html#handling-concurrency>

**Reply by xubobbs** [reply]
I make a server process writed in go . and many sql db opened with ACCESS_MODE=readonly. and only one goroutine write the database . we do test concurrent read and write  this week .and found it always crash in our performance test  and this test only take a few minutes and always crashed 
I see the https://duckdb.org/docs/connect/concurrency.html#handling-concurrency ,and not find the root cause .
so my querstion is : this work way can work well or not ?

**Reply by xubobbs** [reply]
only one go routine to do query and one go routine to write the database at same time and always crashed...

**Comment by xubobbs** [comment]
I use only one connector ,and other conn is derived from this connector ,it seems work now.

**Reply by Tishj** [reply]
I was under the impression you were interacting with the db from multiple processes.
Perhaps you should raise an issue on the go repository if you think there's a bug in concurrent reading/writing behavior with that client.
